tests/integration/route_test.py

```diff
@@ -140,6 +141,67 @@ def test_add_static_route_without_next_hop_address(eth1_up):
     assert_routes(routes, cur_state)
 
 
+@pytest.mark.tier1
+@pytest.mark.skipif(
+    nm_minor_version() < 42,
+    reason="Loopback is only support on NM 1.42+, and blackhole type route "
+    "is stored in loopback",
+)
+def test_add_static_route_with_route_type(eth1_up):
+    route = [
+        {
+            Route.DESTINATION: IPV4_TEST_NET1,
+            Route.ROUTETYPE: Route.ROUTETYPE_BLACKHOLE,
+        },
+        {
+            Route.DESTINATION: IPV6_TEST_NET1,
+            Route.ROUTETYPE: Route.ROUTETYPE_UNREACHABLE,
+        },
+        {
+            Route.DESTINATION: "198.51.100.0/24",
+            Route.ROUTETYPE: Route.ROUTETYPE_PROHIBIT,
+        },
+    ]
+    libnmstate.apply(
+        {
+            Interface.KEY: [ETH1_INTERFACE_STATE],
+            Route.KEY: {Route.CONFIG: route},
+        }
+    )
+    routes_output4 = _get_routes_from_iproute(4, "main")
+    routes_output6 = _get_routes_from_iproute(6, "main")
+    assert IPV4_TEST_NET1 in routes_output4
+    assert Route.ROUTETYPE_BLACKHOLE in routes_output4
+    assert "198.51.100.0/24" in routes_output4
+    assert Route.ROUTETYPE_PROHIBIT in routes_output4
+    assert IPV6_TEST_NET1 in routes_output6
+    assert Route.ROUTETYPE_UNREACHABLE in routes_output6
+
+
+@pytest.mark.tier1
+@pytest.mark.skipif(
+    nm_minor_version() < 42,
+    reason="Loopback is only support on NM 1.42+, and blackhole type route "
+    "is stored in loopback",
+)
+@pytest.mark.tier1
+def test_route_type_with_next_hop_interface(eth1_up):
+    route = [
+        {
+            Route.DESTINATION: IPV4_TEST_NET1,
+            Route.NEXT_HOP_INTERFACE: "eth1",
+            Route.ROUTETYPE: Route.ROUTETYPE_BLACKHOLE,
+        },
+    ]
+    state = {
+        Interface.KEY: [ETH1_INTERFACE_STATE],
+        Route.KEY: {Route.CONFIG: route},
+    }
+
+    with pytest.raises(NmstateNotImplementedError):
```

Comment:
Should be `NmstateValueError` as this is incorrect desired state.

Reply:
I changed into `NmstateValueError`.